src/libs/feature-previews-config.ts

```diff
@@ -4,7 +4,41 @@ export const HAIL_BATCH_AZURE_FEATURE_ID = 'hail-batch-azure';
 export const WORKFLOWS_TAB_AZURE_FEATURE_ID = 'workflows-tab-azure';
 export const DOCKSTORE_AZURE_FEATURE_ID = 'dockstore-azure';
 
-const featurePreviewsConfig = [
+export type FeaturePreview = {
+  /**
+   * ID for the feature. This is used to check if the feature is enabled and to toggle it enabled/disabled.
+   */
+  readonly id: string;
+
+  /**
+   * Name of the feature. Shown on the feature previews page.
+   */
+  readonly title: string;
+
+  /**
+   * Description for the feature. Shown on the feature previews page.
+   */
+  readonly description: string;
+
+  /**
+   * Optional list of groups. If specified, the feature will only appear on the feature previews page
+   * for users that are a member of at least one of the specified groups.
+   * This only applies in production. In dev environments, all features are available to all users.
```

Comment:
This is awesome, thank you! Is it possible to specify in typescript "can be null, but if it's not null then it must have at least one element"? Because the naive reading of the semantics is

- Null: No group requirement
- At least one group listed: the user must be in at least one listed group
- Zero-element list: the user must be in at least one listed group, out of zero, so the feature can never be activated... which feels like an invalid way to configure things?

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4065/commits/dc3e714160b146002032cde9b8d86ecbe9da1df1.